### PR TITLE
Task 11.2

### PR DESCRIPTION
**Pull request by BartekOgien**
Please check

### CODE REVIEW

File: kodilla-patterns/src/main/java/com/kodilla/patterns/prototype/library/Library.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
+package com.kodilla.patterns.prototype.library;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public final class Library extends Prototype {
+
+    String name;
```
Comment (MARudnicki) [inline kodilla-patterns/src/main/java/com/kodilla/patterns/prototype/library/Library.java]: private String name

File: kodilla-patterns/src/main/java/com/kodilla/patterns/prototype/library/Library.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    Set<Book> books = new HashSet<>();
```
Comment (MARudnicki) [inline kodilla-patterns/src/main/java/com/kodilla/patterns/prototype/library/Library.java]: private Set ...

File: kodilla-patterns/src/main/java/com/kodilla/patterns/prototype/library/Library.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    }
+
+    public Library shallowCopy() throws CloneNotSupportedException {
+        return (Library) super.clone();
+    }
+
+    public Library deepCopy() throws CloneNotSupportedException {
+        Library clonedLibrary = (Library)super.clone();
+        clonedLibrary.books = new HashSet<>();
+        for(Book theSet : books) {
```
Comment (MARudnicki) [inline kodilla-patterns/src/main/java/com/kodilla/patterns/prototype/library/Library.java]: Book theSet ???
Book book : books

File: kodilla-patterns/src/main/java/com/kodilla/patterns/prototype/library/Library.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+            clonedLibrary.books.add(theSet);
+        }
```
Comment (MARudnicki) [inline kodilla-patterns/src/main/java/com/kodilla/patterns/prototype/library/Library.java]: Zamiast
        for(Book theSet : books) {
            clonedLibrary.books.add(theSet);
        }
Mozna zapisac ktorej
clonedLibrary.books.addAll(books);

**Review by MARudnicki — CHANGES_REQUESTED** [review]

**Comment by BartekOgien** [comment]
Poprawione

**Pull request merged**